include/libswiftnav/cnav_msg.h

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (C) 2016 Swift Navigation Inc.
+ * Contact: Valeri Atamaniouk <valeri@swift-nav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#ifndef LIBSWIFTNAV_CNAV_MSG_H
+#define LIBSWIFTNAV_CNAV_MSG_H
+
+#include <libswiftnav/common.h>
+#include <libfec/fec.h>
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <limits.h>
+
+/** \addtogroup GPS_L2
+ * \{ */
+/** \addtogroup gps_cnav_decoder
+ * \{ */
+
+/** Size of the Viterbi decoder history. */
+#define GPS_L2_V27_HISTORY_LENGTH_BITS 64
+/** Bits to accumulate before decoding starts. */
+#define GPS_L2C_V27_INIT_BITS    (64)
+/** Bits to decode at a time. */
+#define GPS_L2C_V27_DECODE_BITS  (32)
+/** Bits in decoder tail. We ignore them. */
+#define GPS_L2C_V27_DELAY_BITS   (32)
+/**
+ * GPS CNAV message container.
+ */
+typedef struct
+{
+  u8   prn;    //!< SV PRN. 0..31
+  u8   msg_id; //!< Message id. 0..31
+  u32  tow;    //!< GPS ToW in 6-second units. Multiply to 6 to get seconds.
+  bool alert;  //!< CNAV message alert flag
+} cnav_msg_t;
+
+/**
+ * GPS CNAV decoder component.
+ * This component controls symbol decoding string.
+ *
+ * @sa cnav_msg_decoder_t
+ */
```

Comment:
Comments and documentation are great! Just a small note that we use a slightly different style for our Doxygen comments. Not a big deal but good to be consistent.

https://swiftnav.hackpad.com/Engineering-Playbook-fKcJhOl85sD#:h=Doxygen